flow/src/org/labkey/flow/controllers/WorkspaceData.java

```diff
@@ -200,7 +201,7 @@ else if (path.endsWith(".zip"))
 
                 _object = readWorkspace(file, path);
 
-                FlowProtocol protocol = FlowProtocol.getForContainer(container);
+                FlowProtocol protocol = FlowProtocol.ensureForContainer(user,  container);
```

Comment:
Any problem with this validate method now having a side effect that writes to the database making this not useable for read-only scenarios?